libs/datamodel/core/src/ast/reformat/reformatter.rs

```diff
@@ -42,6 +50,40 @@ impl<'a> Reformatter<'a> {
         Ok(result)
     }
 
+    fn find_all_missing_directives(
+        schema_string: &str,
+    ) -> Result<Vec<MissingFieldDirective>, crate::error::ErrorCollection> {
+        let schema_ast = crate::parse_schema_ast(&schema_string)?;
+        let datamodel = crate::parse_datamodel_and_ignore_datasource_urls(&schema_string)?;
+        let lowerer = crate::transform::dml_to_ast::LowerDmlToAst::new();
+        let mut missing_field_directives = Vec::new();
+        for model in datamodel.models() {
+            let ast_model = schema_ast.find_model(&model.name).unwrap();
+            for field in model.fields() {
+                let original_ast_field = ast_model.fields.iter().find(|f| f.name.name == field.name());
+                let new_ast_field = lowerer.lower_field(&field, &datamodel)?;
+                if original_ast_field.is_some() {
+                    for directive in new_ast_field.directives {
+                        if original_ast_field
+                            .unwrap()
```

Comment:
In Rust you usually try to avoid calling `unwrap()`. Try to make use of an `if let` to avoid it :-)